server/handler.go

```diff
@@ -1485,7 +1485,7 @@ func (h *jobsInsertHandler) Handle(ctx context.Context, r *jobsInsertRequest) (*
 func (h *jobsInsertHandler) addQueryResultToDynamicDestinationTable(ctx context.Context, tx *connection.Tx, r *jobsInsertRequest, response *internaltypes.QueryResponse) error {
 	projectID := r.project.ID
 	jobID := r.job.JobReference.JobId
-	datasetID := jobID
+	datasetID := "ds_" + jobID
```

Comment:
why this is needed?

Reply:
`datasetId` must be unique although `tableId` does not need to. That's why we needed.

reference:
https://cloud.google.com/ruby/docs/reference/google-cloud-bigquery/latest/Google-Cloud-Bigquery-Project#Google__Cloud__Bigquery__Project_create_dataset_instance_
https://cloud.google.com/ruby/docs/reference/google-cloud-bigquery/latest/Google-Cloud-Bigquery-Dataset#Google__Cloud__Bigquery__Dataset_create_table_instance_

Reply:
I think the jobID is already unique for each request. So if you just want to make datasetID unique, jobID seems fine.

Reply:
It seems fine, but actually, after deleting this `ds_` I got an error below. Do you know what causes this erorr? @goccy 

```
2023-09-21 21:49:45 bq-emulator-test-bq-1  | 2023-09-21T12:49:45.555Z   ERROR   server/handler.go:2355   internalError   {"error": "internalError: failed to query SELECT * FROM `job_JSsjGMduSIBf5L_poPWkkL2vZdkd`: no such table: test_job_JSsjGMduSIBf5L_poPWkkL2vZdkd"}
```

Reply:
@kromiii Sorry, I don't known by this information. I'd like to have reproducible test code. Go is preferred, but other languages are also acceptable. Also, please attach the logs obtained at that time.

Reply:
I created a repo to this code. https://github.com/kromiii/bq-emulator-test
This issue depends on the ruby client, so I attached a simple ruby script for testing bigquery-emulator.

Reply:
@prabhav007 Let us know the reason of `ds_` if you are available.

Reply:
In case of jobs with anonymous tables, when the dataset name and the table name is the same, an error response is received. When they are different, it works fine.  PFB the error response I got locally.

http://localhost:9050/bigquery/v2/projects/demo-0d168064/datasets/0ffd9b42-7541-4317-ad16-fad5ec3beb55/tables/0ffd9b42-7541-4317-ad16-fad5ec3beb55/data?maxResults=10&prettyPrint=false

{"error":{"errors":[{"reason":"internalError","location":"","debugInfo":"","message":"failed to query SELECT * FROM `0ffd9b42-7541-4317-ad16-fad5ec3beb55`: no such table: demo-0d168064_0ffd9b42-7541-4317-ad16-fad5ec3beb55"}],"code":500,"message":"failed to query SELECT * FROM `0ffd9b42-7541-4317-ad16-fad5ec3beb55`: no such table: demo-0d168064_0ffd9b42-7541-4317-ad16-fad5ec3beb55"}}

Reply:
@prabhav007 Thank you. I could reproduce the error. However, the quetion is that it works well in the python clinent even if the dataset name and the table name is the same. What makes this difference?

Reply:
I am not sure, I was using the java client for spring.